internal/tools/k8s/secret.go

```diff
@@ -0,0 +1,63 @@
+package k8s
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateSecret creates kubernetes secret.
+func (kc *KubeClient) CreateSecret(namespaceName, secretName, secretKey, secretValue string) (*corev1.Secret, error) {
+	// Check if secret exists first.
+	svc, err := kc.Clientset.CoreV1().Secrets(namespaceName).Get(secretName, metav1.GetOptions{})
+	if err == nil {
+		return svc, nil
+	}
+
+	svcSpec := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespaceName,
+		},
+		StringData: map[string]string{
+			secretKey: secretValue,
+		},
+	}
+	return kc.Clientset.CoreV1().Secrets(namespaceName).Create(svcSpec)
+}
+
+// GetSecrets returns a list of kubernetes namespace objects if they exist.
+//
+// Any errors will be returned immediately and the remaining secrets will be
+// skipped.
+func (kc *KubeClient) GetSecrets(nameSpace string, secretsNames []string) ([]*corev1.Secret, error) {
+	secrets := []*corev1.Secret{}
+	for _, secretsName := range secretsNames {
+		secret, err := kc.Clientset.CoreV1().Secrets(nameSpace).Get(secretsName, metav1.GetOptions{})
+		if err != nil {
+			return secrets, err
+		}
+		secrets = append(secrets, secret)
+	}
+
+	return secrets, nil
+}
+
+// DeleteSecrets deletes kubernetes secrets.
+//
+// Any errors will be returned immediately and the remaining secrets will be
+// skipped.
+func (kc *KubeClient) DeleteSecrets(nameSpace string, secretsNames []string) error {
+	for _, secretName := range secretsNames {
+		err := kc.DeleteSecret(nameSpace, secretName)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeleteSecret delete the given secret.
```

Comment:
I don't think we need this function. We should just directly use the `kc.Clientset` instead of wrapping it like we are doing here.

One of my open PRs is doing that elsewhere to clean things up.

Reply:
It is similar to the namespace.go
not clear to me your comment.

Reply:
I am just saying we should delete this function entirely.

In another PR, I am doing the same to namespace.go and all the other functions that wrap a single Clientset command.

Reply:
done